### PR TITLE
backend/ssm: minor cleanups

### PR DESCRIPTION
**Pull request by rogpeppe**
Avoid the opaque-looking generated mock code and apply some
other minor cleanups.

### CODE REVIEW

File: backend/ssm/ssm.go [hunk 1]
```diff
@@ -6,20 +6,23 @@ import (
...
 	cache   map[string][]byte
 }
 
-func NewBackend(ssm ssmiface.SSMAPI, path string) *Backend {
-	return &Backend{client: ssm, ssmPath: path}
+// NewBackend returns a backend instance that uses the given SSMAPI implementation
+// to retrieve keys from the parameter store at the given path.
+func NewBackend(ssm ssmiface.SSMAPI, path string) backend.Backend {
+	return &ssmBackend{client: ssm, ssmPath: path}
 }
```
Comment (tealeg) [inline backend/ssm/ssm.go]: Public methods should have documentation strings.

File: backend/ssm/ssm.go [hunk 2]
```diff
@@ -30,11 +33,11 @@ func (b *Backend) Get(ctx context.Context, key string) ([]byte, error) {
 	return b.fromCache(ctx, key)
 }
```
Comment (tealeg) [inline backend/ssm/ssm.go]: Same for this one (I know you didn't create this @rogpeppe, I just think it should be addressed)

File: backend/ssm/ssm.go (hunk 1 above)
Comment (rogpeppe) [reply backend/ssm/ssm.go]: done

File: backend/ssm/ssm.go (hunk 2 above)
Comment (rogpeppe) [reply backend/ssm/ssm.go]: done

**Pull request merged**